### PR TITLE
use RaftEngine instead of RocksEngine in server module

### PR DESCRIPTION
**Pull request by hicqu**
### What problem does this PR solve?

Use RaftEngine instead of RocksEngine in server module.

### Check List 

Tests 

- Unit test
- Integration test

Side effects

### Release note 
* No release note

### CODE REVIEW

File: components/engine_traits/src/metrics_flusher.rs [hunk 1]
```diff
@@ -49,7 +49,7 @@ impl<K: KvEngine, R: RaftEngine> MetricsFlusher<K, R> {
                     raft_db.flush_metrics("raft");
                     if last_reset.elapsed() >= FLUSHER_RESET_INTERVAL {
                         kv_db.reset_statistics();
-                        raft_db.reset_statistics();
+                        // raft_db.reset_statistics();
```
Comment (NingLin-P) [inline components/engine_traits/src/metrics_flusher.rs:52]: Should remove this line.

**Review by NingLin-P — APPROVED** [review]
Rest LGTM

**Review by hunterlxt — APPROVED** [review]
Most LGTM

File: components/raftstore/src/router.rs [hunk 2]
```diff
@@ -33,6 +33,8 @@ where
         msg: SignificantMsg<EK::Snapshot>,
     ) -> RaftStoreResult<()>;
 
+    fn broadcast_normal(&self, msg_en: impl FnMut() -> PeerMsg<EK>);
+
     fn send_casual_msg(&self, region_id: u64, msg: CasualMessage<EK>) -> RaftStoreResult<()> {
```
Comment (hunterlxt) [inline components/raftstore/src/router.rs:38]: not related to this PR, but it's better to complete these comments

**Pull request merged**